src/middlewared/middlewared/plugins/reporting/events.py

```diff
@@ -91,6 +91,9 @@ def run_sync(self):
             data['cpu']['temperature_celsius'] = self.middleware.call_sync('reporting.cpu_temperatures')
             data['cpu']['temperature'] = {k: 2732 + int(v * 10) for k, v in data['cpu']['temperature_celsius'].items()}
 
+            # Flag telling whether the CPU temperatures' data is available
+            data['cpu_temperatures_retrieved'] = bool(sum(v for v in data['cpu']['temperature_celsius'].values()))
```

Comment:
The name of the key doesn't match your comment. I'd rather name it `_available` to reflect reality.
```suggestion
            data['cpu_temperatures_available'] = bool(sum(v for v in data['cpu']['temperature_celsius'].values()))
```

Finally, there has to be a better way of knowing whether or not we have cpu temperatures then doing `sum` again since we're storing this information in the `['cpu']['temperature_celsius']` key up above, right?